src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -16,40 +16,88 @@
 
 package io.supertokens.oauth;
 
+import com.google.gson.JsonObject;
 import io.supertokens.Main;
 import io.supertokens.config.Config;
+import io.supertokens.httpRequest.HttpRequest;
+import io.supertokens.httpRequest.HttpResponseException;
+import io.supertokens.multitenancy.Multitenancy;
+import io.supertokens.oauth.exceptions.OAuthException;
 import io.supertokens.pluginInterface.Storage;
 import io.supertokens.pluginInterface.StorageUtils;
 import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
 import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.pluginInterface.oauth.OAuthAuthResponse;
 import io.supertokens.pluginInterface.oauth.OAuthStorage;
 
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class OAuth {
 
-    public static void getAuthorizationUrl(Main main, AppIdentifier appIdentifier, Storage storage, String clientId,
-                                           String redirectURI, String responseType, String scope, String state)
-            throws InvalidConfigException {
+    private static final String LOCATION_HEADER_NAME = "Location";
+    private static final String COOKIES_HEADER_NAME = "Set-Cookie";
+
+    public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier appIdentifier, Storage storage, String clientId,
+                                                        String redirectURI, String responseType, String scope, String state)
+            throws InvalidConfigException, HttpResponseException, IOException, OAuthException, StorageQueryException,
+            TenantOrAppNotFoundException {
         // TODO:
         // - validate that client_id is present for this tenant
-        // - call hydra
-        // - if location header is:
-        //     - localhost:3000, then we redirect to apiDomain
-        //     - public url for hydra, then we throw a 400 error with the right json
-        //     - else we redirect back to the client
 
         OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
 
         String redirectTo = null;
+        List<String> cookies = null;
+
+        String publicOAuthProviderServiceUrl = Config.getBaseConfig(main).getOAuthProviderPublicServiceUrl();
```

Comment:
if this is null (not configured by the user), we need to throw an exception like BadInputException (which will eventually yield a 400 error from the core).

Reply:
Also, using getBaseConfig(main) is not the right thing todo, since that does not load the config for the current app that's querying this API. Instead, you want to use `Config.getConfig(appIdentifier.toTenantIdentifier())` (or whatever the function name is called in the AppIdentifier class)

Reply:
If that's null, the getOAuthProviderPublicServiceUrl will throw an InvalidConfigException

Reply:
> Also, using getBaseConfig(main) is not the right thing todo, since that does not load the config for the current app that's querying this API. Instead, you want to use `Config.getConfig(appIdentifier.toTenantIdentifier())` (or whatever the function name is called in the AppIdentifier class)

Done